### PR TITLE
Remove autoreconnect feature. It is handled by socket.io.

### PR DESCRIPTION
**Pull request by ruffrey**
Web socket connections were leaking when auth type was App-Secret. This is
because we added autoreconnect code and it duplicated what socket.io was
doing. I am unsure why this was not clear at the time it was implemented.
Nonetheless, we fall back to using socket.io's reconnection logic, but
only when the web socket was initiated using an App-Secret (that token
lives forever).

Because endpoint brokered auth might work sometimes but fail most of the
time due to the App-Token expiring, we explicitly forbid App-Token web
sockets from using socket.io's reconnect logic. So they will not reconnect
at all - it is an implementation detail left to the user of the library.
They would have needed to obtain an auth token for the initial connection
anyways, so presumably on disconnect they would repeat that process - like
one would do with respoke.js for the browser.

fix: MER-4681

Consider a major version bump due to different behavior for brokered
endpoint auth clients. On the other hand, it is fixing definitely-broken
behavior, so maybe this can be a minor bump.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -656,29 +586,24 @@ function Client(options) {
...
             self.socket.on('reconnect', function (num) {
                 debug('event reconnect', self.endpointId, num);
-                /**
-                 * Reconnected with respoke.
-                 * @event reconnect
-                 * @property {number} num
-                 */
-                self.emit('reconnect', num);
-                self.clearAutoreconnectTimeout();
+                self.wsCall('post', '/connections', dataBody)
```
Comment (leedm777) [inline lib/client.js]: extract a function that both `on('reconnect'...` and `once('connect'...` call to reduce duplication

**Comment by ruffrey** [comment]
I tested this locally using our internal respoke connection-test suite that runs everything in docker-compose, stopping and starting the respoke backend several times. It no longer leaks connections, and the brokered auth endpoint web socket will *no longer* reconnect (as expected).

File: lib/client.js (hunk 1 above)
Comment (ruffrey) [inline lib/client.js]: Done!

**Comment by ruffrey** [comment]
After conferring with @leedm777 we decided version numbers are cheap, and we technically removed `autoreconnect` as an option, so we are going to do a major bump.

**Pull request merged**